backend/login/tests.py

```diff
@@ -96,42 +132,38 @@ def test_existing_email_directly(self):
         # Assertions
         self.assertIsInstance(response, Response)
         self.assertEqual(response.status_code, 200)
-        self.assertEqual(response.data, {"message": f"Got user's email {email}", "data": "21"})
-        self.mock_use_case.match_email_to_id.assert_called_once_with(email)
+        self.assertEqual(
+            response.data,
+            {
+                "message": f"Got user's email {request.data}",
+                "data": "21"
+            }
+        )
+        self.mock_use_case.match_email_to_id.assert_called_once_with(request.data)
+        self.assertEqual(request.session["user_id"], "21")
 
     def test_non_existing_email_directly(self):
         """
         Test the logic inside the view directly with a non-existing email.
         """
-        email = {"email": "nonexistentemail@gmail.com"}
-        django_request = self.factory.post("/login/get_email/", data=email, content_type="application/json")
-        
-        # Wrap Django's request with DRF's Request
-        request = Request(django_request)
+        email_data = {"userEmail": "nonexistentuser@example.com"}
+        request = self._setup_request(email_data)
```

Comment:
so self._setup_request is the fix to the bug? I assume this function will set up the mock front end request.

Reply:
Great work on figuring this out!!